### ISSUE TITLE
voice assistant "ok google"

### ISSUE THREAD

**Issue by markusuidl** [openhab/openhab-android#2394]
I've been dealing with voice input solutions for a long time, but none of them convinced me. Either you need an additional device (Google Home, Alexa, etc ...) or it runs via an opensource solution, which is complicated to install or simply does not offer good detection.

In my opinion, you usually have your smartphone with you or within easy reach.

It would be very cool if you could use the google assistant with "ok google" for voice input in openhab.

Currently it is about the widget, but the device has to be on and you have to operate the widget before you can talk.

**I am thinking of the following solution:**

Signal words can be registered in the Google Assistant via the Google API. A signal word could be "with openhab".

So you could send every text to openhab and there, in a rule, parse the text and carry out the corresponding action.

some examples:
- OK Google - turn on the TV with Openhab.
- OK Google - Open the shutters in the living room to 50% with openhab.

Thanks to the device prefix, which is already implemented, different actions can be triggered depending on the user.

**Comment by rkoshak** [comment]
> It would be very cool if you could use the google assistant with "ok google" for voice input in openhab.

I'm confused. This is exactly what the Google Assistant integration already provides. https://www.openhab.org/docs/ecosystem/google-assistant/

You don't even need to register a "talk to" or "with" set of keywords. Once set up and once your Items are tagged your openHAB stuff will appear in the Google Home app and can be controlled by name in Google Assistant. You can even use it with some easter eggs. I just wowed my son this morning by saying "Hey google, lumos" (Harry Potter reference) and all my openHAB lights turned ON. 

In short, you can already do what you are asking for without the "with openHAB" part.

But if there are things you want to change about how that currently works, it won't be the Android app that implements it. It will be the Google Assistant integration that does it.

**Comment by mueller-ma** [comment]
As @rkoshak said this is already possible with Google Assistant.
